### PR TITLE
Fread/fgets

### PR DESCRIPTION
**Pull request by fennecdjay**
Here it is :tada:

### CODE REVIEW

File: zpm.c [hunk 1]
```diff
@@ -85,12 +85,15 @@ int generate_plugin_entry(char* plugin_name) {
...
     char* plugin_entry = get_plugin_entry(plugin_name);
     char  plugin_entry_list[PATH_MAX];
-    fread(plugin_entry_list, 1, PATH_MAX, store);
-    fclose(store);
-    if (strstr(plugin_entry_list, plugin_entry)) {
-        free(plugin_entry);
-        return 0;
+
+    while (fgets(plugin_entry_list, PATH_MAX, store)) {
+            if (strstr(plugin_entry_list, plugin_entry)) {
```
Comment (desyncr) [inline zpm.c]: Minor indentation issue here.

File: zpm.c [hunk 2]
```diff
@@ -136,9 +139,10 @@ int plugin_list_add_item(char* plugin_name) {
     }
     strcat(plugin_item, "\n");
 
-    fread(plugin_item_list, 1, PATH_MAX, store);
-    if (strstr(plugin_item_list, plugin_item)) {
-        return 0;
+    while (fgets(plugin_item_list, PATH_MAX, store)) {
+        if (strstr(plugin_item_list, plugin_item)) {
+            return 0;
```
Comment (desyncr) [inline zpm.c]: Aren't we missing a fclose(init) here, and after fwrite?

**Review by desyncr — CHANGES_REQUESTED** [review]
I always forget to submit the review!

File: zpm.c (hunk 2 above)
Comment (fennecdjay) [reply zpm.c]: Good catch!!!
That's why we should test *everything* with valgrind 😅

**Review by desyncr — COMMENTED** [review]
Done!

**Pull request merged**